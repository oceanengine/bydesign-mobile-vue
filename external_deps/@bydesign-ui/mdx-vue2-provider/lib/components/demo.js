/**
 * Component Demo
 */
import createBEM from '../utils/bem';
import HL from './highlight';
import { openCodepen, openSandbox, copyToClipboard } from '../utils/editor';
var bem = createBEM('demo');

var Demo = function Demo() {
  return {
    props: {
      title: [String, Number],
      desc: [String, Number],
      code: String,
      src: String,
      srcRender: Function,
      maxHeight: Number
    },
    data: function data() {
      return {
        unfolded: false
      };
    },
    methods: {
      setUnfolded: function setUnfolded(value) {
        this.unfolded = value;
      }
    },
    render: function render() {
      var h = arguments[0];
      var title = this.title,
          desc = this.desc,
          code = this.code,
          srcRender = this.srcRender,
          unfolded = this.unfolded,
          setUnfolded = this.setUnfolded,
          _this$maxHeight = this.maxHeight,
          maxHeight = _this$maxHeight === void 0 ? 200 : _this$maxHeight;
      var containerStyle = {
        maxHeight: "".concat(maxHeight, "px")
      };
      return h("div", {
        "class": bem()
      }, [h("div", {
        "class": bem('title'),
        "attrs": {
          "id": title.toString()
        }
      }, [title]), srcRender !== undefined && h("div", {
        "class": bem('box')
      }, [h("div", {
        "class": bem('show')
      }, [srcRender()]), desc && h("div", {
        "class": bem('desc')
      }, [desc]), code !== undefined && h("div", {
        "class": bem('code-tool')
      }, [h("div", {
        "class": bem('operations')
      }, [h("span", {
        "class": bem('operation-btn'),
        "on": {
          "click": function click() {
            openCodepen(unescape(code));
          }
        },
        "domProps": {
          "innerHTML": require('!html-loader!../components/icons/codepen.svg')
        }
      }), h("span", {
        "class": bem('operation-btn'),
        "on": {
          "click": function click() {
            openSandbox(unescape(code));
          }
        },
        "domProps": {
          "innerHTML": require('!html-loader!../components/icons/sandbox.svg')
        }
      }), h("span", {
        "class": bem('operation-btn'),
        "on": {
          "click": function click() {
            copyToClipboard(unescape(code));
          }
        },
        "domProps": {
          "innerHTML": require('!html-loader!../components/icons/copy.svg')
        }
      }), h("span", {
        "class": bem('operation-btn'),
        "on": {
          "click": function click() {
            setUnfolded(!unfolded);
          }
        },
        "domProps": {
          "innerHTML": require('!html-loader!../components/icons/code.svg')
        }
      })])]), code !== undefined && unfolded && h("div", {
        "class": bem('code'),
        "style": containerStyle
      }, [h(HL, {
        "attrs": {
          "code": unescape(code)
        }
      })])])]);
    }
  };
};

export default Demo;