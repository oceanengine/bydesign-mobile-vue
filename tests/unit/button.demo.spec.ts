import Demo from '@src/components/button/demo/index.vue';
import { snapshotDemo } from '../demo';

snapshotDemo(Demo);
