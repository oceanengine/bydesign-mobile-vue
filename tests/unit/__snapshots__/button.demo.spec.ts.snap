// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders demo correctly 1`] = `
<div class="demo button">
  <div class="byted-header--wrapper">
    <div class="byted-header byted-header--default byted-header--fixed byted-hairline--bottom">
      <div class="byted-header__left"><i class="byted-header__icon" style="width: 1.2rem; height: 1.2rem;">test-file-stub</i></div>
      <div class="byted-header__title"><span class="demo-header-icon"><img src="https://s2.pstatp.com/cdn/expire-1-d/byted-creative-app/static/ByDesign-logo.svg" class="img-icon"></span></div>
    </div>
  </div>
  <div><span class="byted-icon byted-icon-module-button" style="font-size: 20px;"><svg width="20" height="20" viewBox="0 0 64 64" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M14 8H38.4286H50V24H38.4286H14V8Z" fill="transparent" stroke="#333" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path><path d="M22 16H42" stroke="#333" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path><path fill-rule="evenodd" clip-rule="evenodd" d="M25.9469 50C19.4279 42.8771 15.6873 38.7492 14.7251 37.6162C13.2817 35.9168 13.8878 33.996 17.4999 33.996C21.1121 33.996 23.1953 39.2814 25.9469 39.2814C25.9469 39.2814 25.9644 32.5257 25.9502 19.004C25.9485 17.3467 27.2906 16.0017 28.9479 16L28.9511 16C30.6102 16 31.9553 17.345 31.9553 19.0042V31.0136C39.928 32.2224 44.2626 32.8893 44.9589 33.0144C46.0034 33.2019 47.9469 34.199 47.9469 37.068C47.9469 38.9807 48 45.9581 48 56H25.9502L25.9469 50Z" fill="transparent" stroke="#333" stroke-width="4" stroke-linejoin="round"></path><line x1="27" y1="50" x2="47" y2="50" stroke="#333" stroke-width="4" stroke-linecap="round"></line></svg></span></div>
  <div>
    <div class="byted-button byted-button--default byted-button--large"></div>
    <div class="byted-button byted-button--primary byted-button--large"></div>
    <div class="byted-button byted-button--success byted-button--large"></div>
    <div class="byted-button byted-button--warning byted-button--large"></div>
    <div class="byted-button byted-button--danger byted-button--large"></div>
  </div>
  <div>
    <div class="byted-button byted-button--primary byted-button--large byted-button--plain byted-hairline--surround"></div>
    <div class="byted-button byted-button--success byted-button--large byted-button--plain byted-hairline--surround"></div>
    <div class="byted-button byted-button--warning byted-button--large byted-button--plain byted-hairline--surround"></div>
    <div class="byted-button byted-button--danger byted-button--large byted-button--plain byted-hairline--surround"></div>
  </div>
  <div>
    <div class="byted-button byted-button--primary byted-button--large byted-button--plain byted-button--fade byted-hairline--surround"></div>
    <div class="byted-button byted-button--success byted-button--large byted-button--plain byted-button--fade byted-hairline--surround"></div>
    <div class="byted-button byted-button--warning byted-button--large byted-button--plain byted-button--fade byted-hairline--surround"></div>
    <div class="byted-button byted-button--danger byted-button--large byted-button--plain byted-button--fade byted-hairline--surround"></div>
  </div>
  <div>
    <div class="byted-button byted-button--primary byted-button--large byted-button--plain byted-button--hairline byted-hairline--surround"></div>
    <div class="byted-button byted-button--success byted-button--large byted-button--plain byted-button--hairline byted-hairline--surround"></div>
    <div class="byted-button byted-button--warning byted-button--large byted-button--plain byted-button--fade byted-button--hairline byted-hairline--surround"></div>
    <div class="byted-button byted-button--danger byted-button--large byted-button--plain byted-button--fade byted-button--hairline byted-hairline--surround"></div>
  </div>
  <div>
    <div class="byted-button byted-button--primary byted-button--large byted-button--square"></div>
    <div class="byted-button byted-button--primary byted-button--large byted-button--round"></div>
  </div>
  <div>
    <div class="byted-button byted-button--default byted-button--large byted-button--plain byted-button--disabled byted-hairline--surround"></div>
    <div class="byted-button byted-button--primary byted-button--large byted-button--plain byted-button--disabled byted-hairline--surround"></div>
    <div class="byted-button byted-button--primary byted-button--large byted-button--disabled"></div>
  </div>
  <div>
    <div class="byted-button byted-button--default byted-button--large byted-button--plain byted-hairline--surround">
      <div class="byted-loading byted-button__loading"><i class="byted-loading__icon byted-loading-circle">test-file-stub</i></div>
    </div>
    <div class="byted-button byted-button--primary byted-button--large byted-button--plain byted-hairline--surround">
      <div class="byted-loading byted-button__loading"><i class="byted-loading__icon byted-loading-circle">test-file-stub</i></div>
    </div>
    <div class="byted-button byted-button--primary byted-button--large">
      <div class="byted-loading byted-button__loading"><i class="byted-loading__icon byted-loading-circle">test-file-stub</i></div>
    </div>
    <div class="byted-button byted-button--primary byted-button--midget">
      <div class="byted-loading byted-button__loading"><i class="byted-loading__icon byted-loading-circle">test-file-stub</i></div>
    </div>
  </div>
  <div>
    <div class="byted-button byted-button--text byted-button--large"></div>
  </div>
  <div class="dome-size">
    <div class="dome-size-content"><span class="dome-size-content_title">尺寸</span>
      <div class="dome-size-content_slider">
        <div class="dome-slider">
          <div class="dome-slider-container">
            <div class="popover" style="display: none;"><span class="content">large</span>
              <div class="popover-arrow"></div>
            </div>
            <div class="dome-slider-button" style="left: 100%;"></div>
            <div class="dome-slider-active" style="width: 100%;"></div>
            <div class="slider-stop slider-stop-active" style="left: 0px;"></div>
            <div class="slider-stop slider-stop-active" style="left: 25%;"></div>
            <div class="slider-stop slider-stop-active" style="left: 50%;"></div>
            <div class="slider-stop slider-stop-active" style="left: 75%;"></div>
            <div class="slider-stop slider-stop-active" style="left: 100%;"></div>
          </div>
        </div>
      </div>
    </div>
    <div class="dome-size-icon"><img src="test-file-stub" alt="" class="svg-icon-close" style="display: none;"> <img src="test-file-stub" alt="" class="svg-icon-transfer"></div>
  </div>
</div>
`;
